File: dotnet_3/cs/rest/SCS/Imp/DateParse.cs
namespace SCS.Imp
{
    public class DateParse
    {
        public static string Subject(string dayName, string monthName)
        {
            var result = 0;
            //int month = -1;
            dayName = dayName.ToLower();
            monthName = monthName.ToLower();

            if ("mon".Equals(dayName) ||
                "tue".Equals(dayName) ||
                "wed".Equals(dayName) ||
                "thur".Equals(dayName) ||
                "fri".Equals(dayName) ||
                "sat".Equals(dayName) ||
                "sun".Equals(dayName))
            {
                result = 1;
            }

            if (monthName == "jan")
                result += 1;
            else if (monthName == "feb")
                result += 2;
            else if (monthName == "mar")
                result += 3;
            else if (monthName == "apr")
                result += 4;
            else if (monthName == "may")
                result += 5;
            else if (monthName == "jun")
                result += 6;
            else if (monthName == "jul")
                result += 7;
            else if (monthName == "aug")
                result += 8;
            else if (monthName == "sep")
                result += 9;
            else if (monthName == "oct")
                result += 10;
            else if (monthName == "nov")
                result += 11;
            else if (monthName == "dec") result += 12;

            return "" + result;
        }
    }
}

File: jdk_8_maven/cs/graphql/graphql-scs/src/main/kotlin/org/graphqlscs/type/DateParse.kt
package org.graphqlscs.type

import org.springframework.stereotype.Component
import java.util.*
@Component
class DateParse {
    fun subject(dayname: String, monthname: String): String {
        var dayname = dayname
        var monthname = monthname
        var result = 0
        //int month = -1;
        dayname = dayname.lowercase(Locale.getDefault())
        monthname = monthname.lowercase(Locale.getDefault())
        if ("mon" == dayname || "tue" == dayname || "wed" == dayname || "thur" == dayname || "fri" == dayname || "sat" == dayname || "sun" == dayname) {
            result = 1
        }
        if ("jan" == monthname) {
            result += 1
        }
        if ("feb" == monthname) {
            result += 2
        }
        if ("mar" == monthname) {
            result += 3
        }
        if ("apr" == monthname) {
            result += 4
        }
        if ("may" == monthname) {
            result += 5
        }
        if ("jun" == monthname) {
            result += 6
        }
        if ("jul" == monthname) {
            result += 7
        }
        if ("aug" == monthname) {
            result += 8
        }
        if ("sep" == monthname) {
            result += 9
        }
        if ("oct" == monthname) {
            result += 10
        }
        if ("nov" == monthname) {
            result += 11
        }
        if ("dec" == monthname) {
            result += 12
        }
        return "" + result
    }

}